independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/validation/AbstractNormalScopedFinalTest.java

```diff
@@ -0,0 +1,92 @@
+package io.quarkus.arc.test.validation;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.quarkus.arc.test.ArcTestContainer;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.Vetoed;
+import javax.enterprise.inject.spi.DefinitionException;
+import javax.enterprise.inject.spi.DeploymentException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public abstract class AbstractNormalScopedFinalTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = createTestContainer();
+
+    protected abstract ArcTestContainer createTestContainer();
+
+    @Test
+    public void testFailure() {
+        Throwable error = container.getFailure();
+        assertNotNull(error);
+        assertTrue(error instanceof DeploymentException);
+        assertNotNull(error.getCause());
+        assertTrue(error.getCause() instanceof DefinitionException);
+    }
+
+    @ApplicationScoped
+    static final class Unproxyable {
+
+        void ping() {
+        }
+
+    }
+
+    @ApplicationScoped
+    static class FieldProducerWithFinalClass {
+
+        @Produces
+        @ApplicationScoped
+        public FinalFoo foo = new FinalFoo();
+    }
+
+    @ApplicationScoped
+    static class MethodProducerWithFinalClass {
+
+        @Produces
+        @ApplicationScoped
+        public FinalFoo createFoo() {
+            return new FinalFoo();
+        }
+    }
+
+    @ApplicationScoped
+    static class FieldProducerWithFinalWrongConstructor {
```

Comment:
Why the word `Final` here? I don't see anything final in this case

Reply:
Good catch ;-). It should be probably `FieldProducerWithWrongConstructor`.